app/scripts/controllers/balance.js

```diff
@@ -0,0 +1,60 @@
+const ObservableStore = require('obs-store')
+const PendingBalanceCalculator = require('../lib/pending-balance-calculator')
+const BN = require('ethereumjs-util').BN
+
+class BalanceController {
+
+  constructor (opts = {}) {
+    const { address, ethStore, txController } = opts
+    this.address = address
+    this.ethStore = ethStore
+    this.txController = txController
+
+    const initState = {
+      ethBalance: undefined,
+    }
+    this.store = new ObservableStore(initState)
+
+    this.balanceCalc = new PendingBalanceCalculator({
+      getBalance: () => Promise.resolve(this._getBalance()),
+      getPendingTransactions: this._getPendingTransactions.bind(this),
+    })
+
+    this.registerUpdates()
+  }
+
+  async updateBalance () {
+    const balance = await this.balanceCalc.getBalance()
+    this.store.updateState({
+      ethBalance: balance,
+    })
+  }
+
+  registerUpdates () {
+    const update = this.updateBalance.bind(this)
+    this.txController.on('submitted', update)
+    this.txController.on('confirmed', update)
+    this.txController.on('failed', update)
+    this.txController.blockTracker.on('block', update)
```

Comment:
instead of 
```
// ...
this.txController.on('submitted', update)
this.txController.blockTracker.on('block', update)
```
you could just subscribe to `ethStore` and  `this.txController.memStore`

Reply:
I'm ok w/ the ethStore subscription instead of the blockTracker, but subscribing to specific, relevant txController state changes seems like a more efficient way to pre-filter the data we process, instead of processing on every txController mem change, which includes every single tx update of every sort.